### PR TITLE
Update Micrometer from 0.10.0 to 1.0.0-rc.4

### PR DESCRIPTION
**Pull request by trustin**
- Remove RolldingHdrHistogram
- Remove the NOTICE.txt items and LICENSE files for rolling-metrics and
  HdrHistogram
- Remove ConcurrentCompositeMeterRegistry and its related classes
- Rename MeterId to MeterIdPrefix for disambiguation with Meter.Id
  - Rename MeterIdFunction to MeterIdPrefixFunction
  - Rename MeterIdFunctionFactory to MeterIdPrefixFunctionFactory
- Rename ArmeriaMeterBinders to ArmeriaMeterBindersConfiguration
  following the convention of Micrometer
- Clean up the dependencies of armeria-spring-boot-autoconfigure

### CODE REVIEW

**Comment by trustin** [comment]
Note that this PR is scheduled for 0.56.0, not 0.55.0, so that we can ship the fix for #810 at least once before our users upgrade Micrometer.

**Comment by imasahiro** [comment]
Can you remove ProGuard conf for HdrHistogram as well?
https://github.com/line/armeria/pull/823/files#diff-26fb43127aab27a5a6df14164a6f424aR111

**Comment by trustin** [comment]
Needs investigation:

```
com.linecorp.armeria.spring.ArmeriaMeterBindersConfigurationTest > testDefaultMetrics FAILED
    java.util.ConcurrentModificationException
        at java.util.HashMap$Values.forEach(HashMap.java:983)
        at com.linecorp.armeria.common.metric.MoreMeters.measureAll(MoreMeters.java:80)
        at com.linecorp.armeria.spring.ArmeriaMeterBindersConfigurationTest.testDefaultMetrics(ArmeriaMeterBindersConfigurationTest.java:74)
```

File: NOTICE.txt [hunk 1]
```diff
@@ -93,11 +93,6 @@ This product contains a modified part of Micrometer, distributed by Pivotal Soft
   * License: licenses/LICENSE.micrometer.al20.txt (Apache License v2.0)
```
[the remark this answers is not in the document]
Comment (trustin) [reply NOTICE.txt]: Fixed :-)

**Comment by trustin** [comment]
> Needs investigation

Fixed the test failure.

**Comment by trustin** [comment]
We may want to wait for 1.0.0-rc.4 due to another thread safety issue in 1.0.0-rc.3: https://github.com/micrometer-metrics/micrometer/pull/209

**Comment by trustin** [comment]
Fixed various issues related with Micrometer: https://github.com/micrometer-metrics/micrometer/pulls?q=is%3Apr%20author%3Atrustin A new version should be released soon.

**Comment by trustin** [comment]
Updated to 1.0.0-rc.4 with additional changes:

- Rename MeterId to MeterIdPrefix for disambiguation with Meter.Id
  - Rename MeterIdFunction to MeterIdPrefixFunction
  - Rename MeterIdFunctionFactory to MeterIdPrefixFunctionFactory

**Comment by trustin** [comment]
Thanks, reviewers!

**Pull request merged**